packages/plexus/README.md

```diff
@@ -1,13 +1,914 @@
+
+| Name | Type and description |
+| :-- | :-- |
+| setOnNode | `TNodePropsSetter` |
+|  | Allows props to be defined or generated for the container of the node. This is a `<div>` for HTML layers and a `<g>` for SVG layers. **Note:** The resultant props are applied to the container element; they are not passed on to the `renderNode` factory.<br>&nbsp; |
+| renderNode | `TRenderNodeFn` |
+|  | **Required**<br>A factory function that is used to generate nodes from the `TLayoutVertex` data. `renderNode` is invoked for each `TLayoutVertex`. Unlike measurable nodes layers, the `TLayoutVertex` will always be available. This function will have access to the `TRenderUtils`, which means it can access the current zoom transform, but it is not redrawn when the zoom transform changes.<br>&nbsp; |
+
+The types for `setOnNode` and `renderNode` are distinct from the corresponding fields on a measurable nodes layer in that the `layoutVertex` argument is **always available**.
+
+The type for `setOnNode` is similar to that of `setOnContainer` in that the value can also be either an object of props, a factory function, or an array of either.
+
+```tsx
+type TNodePropsSetter = Record<string, unknown> | TNodesPropsFn | (TNodesPropsFn | Record<string, unknown>)[];
+
+type TNodesPropsFn = (
+  layoutVertex: TLayoutVertex | null,
+  utils: TRendererUtils
+) => Record<string, unknown> | null;
+```
+
+The type for the `renderNode` field.
+
+```tsx
+type TRenderNodeFn = (layoutVertex: TLayoutVertex, utils: TRendererUtils) => React.ReactNode;
+```
+
+#### Edges layer
+
+Any number of edges layers can be configured for a `LayoutDigraph` (TODO: rename). Edges layers are more restrictive (or less mature) than nodes layers, at present:
+
+- The `layerType` of edges layers must be `"svg"`
+- Edges layers do not afford a `renderEdge` equivalent to the `renderNode`.
+
+Thus, edges are less configurable than nodes (for now). If you need additional functionality, please [file a ticket](https://github.com/jaegertracing/jaeger-ui/issues/new?labels=plexus&title=Plexus+edge+enhancements).
+
+The builtin renderer for edges draws a path based on the `pathPoints` field of the `TLayoutEdge` being rendered.
+
+In addition to the common layer configuration fields, the following fields are also available:
+
+| Name | Type and description |
+| :-- | :-- |
+| edges | `true` |
+|  | **Required**<br>Indicates the layer will render edges.<br>&nbsp; |
+| layerType | `"svg"` |
+|  | **Required**<br>Edges must be SVG layers.<br>&nbsp; |
+| markerEndId | `string` |
+|  | This field should refer to an element to use as the [`marker-end`](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/marker-end) for the edge `<path>`. A typical scenario matches the `markerEndId` to the `localId` of a `TDefEntry`. Each of the `localId` and `markerEndId` are passed through the `getLocalId()` util (TODO: reaname the util) to generate an ID unique within the document.<br>&nbsp; |
```

Comment:
what would the util be called instead of `getLocalId`?

Reply:
`getGlobalId` (fixed in another PR)

Reply:
I see. the arg is the "local identifier" and the return value is supposed to be globally unique